Mathlib/GroupTheory/FreeAbelianGroup.lean

```diff
@@ -68,7 +68,23 @@ universe u v
 
 variable (α : Type u)
 
-/-- The free abelian group on a type. -/
+/--
+If `α` is a type, then `FreeAbelianGroup α` is the free abelian group generated by `α`.
+This is an abelian group equipped with a function
+`FreeAbelianGroup.of : α → FreeAbelianGroup α` which has the following universal property:
+if `G` is any abelian group, and `f : α → G` is any function, then this function is
+the composite of `FreeAbelianGroup.of` and a unique group homomorphism
+`FreeAbelianGroup.lift f : FreeAbelianGroup α →+ G`.
+
+A typical element of `FreeAbelianGroup α` is a formal sum of
+elements of `α` and their formal inverses.
+For example if `x` and `y` are terms of type `α` then `x + x + x - y` is a
+"typical" element of `FreeAbelianGroup α`. In particular if `α` is empty
+then `FreeAbelianGroup α` is isomorphic to the trivial group, and if `α` has one term
+then `FreeAbelianGroup α` is isomorphic to `ℤ`.
```

Comment:
You could add here that elements of `FreeAbelianGroup α` can be thought of as functions `α → ℤ` which have finite support.